### PR TITLE
Introduce customizable hotkeys #1728

### PR DESCRIPTION
**Pull request by christianguenter2**
With this commit customizable hotkeys are introduced. They can
be defined in the user specific settings. Currently we support
only one letter keybindigs e.g. 's', 't' or 'x' and no modifiers.

Every page in the UI can decide which actions it offers for the
hotkeys. Therefore we introduce the interface ZIF_ABAPGIT_GUI_PAGE_HOTKEY
with the method GET_HOTKEY_ACTIONS. At this point in time only
the main page offers actions for hotkeys, but the mechanism works
already for all pages. New actions for hotkeys can be defined
easily.

Hotkeys are only available for installed abapGit repositories.
Because we need to detect the classes which implement the above
mentioned interface and it seems that there is no easy way to do
that for local classes. Maybe we can add it later when we know more.

We don't supply default key bindings. So it is totally up to the
user to define them.

image

#1728

### CODE REVIEW

**Comment by larshp** [comment]
will this work for local classes?

File: src/ui/zcl_abapgit_gui_page_main.clas.abap [hunk 1]
```diff
@@ -351,4 +352,40 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_MAIN IMPLEMENTATION.
...
+    ls_hotkey_action-action = zif_abapgit_definitions=>gc_action-repo_newonline.
+    INSERT ls_hotkey_action INTO TABLE rt_hotkey_actions.
+
+    ls_hotkey_action-name   = |Main: Uninstall|.
+    ls_hotkey_action-action = zif_abapgit_definitions=>gc_action-repo_purge.
+    INSERT ls_hotkey_action INTO TABLE rt_hotkey_actions.
+
+    ls_hotkey_action-name   = |Main: Show diff|.
+    ls_hotkey_action-action = zif_abapgit_definitions=>gc_action-go_diff.
+    INSERT ls_hotkey_action INTO TABLE rt_hotkey_actions.
```
Comment (larshp) [inline src/ui/zcl_abapgit_gui_page_main.clas.abap]: no default key per action? I use abapGit across multiple systems, would like not to set it up on each system

Comment (christianguenter2) [reply src/ui/zcl_abapgit_gui_page_main.clas.abap]: added default hotkeys

**Comment by larshp** [comment]
Any way to list the configured hotkeys, like pressing `?` on github? Anyhow thats an idea for a follow up PR

**Pull request merged**